Source/WTF/wtf/NativePromise.h

```diff
@@ -792,42 +789,40 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
             if (ThenCallbackBase::m_targetQueue)
                 assertIsCurrent(*ThenCallbackBase::m_targetQueue);
             ASSERT(m_settleFunction);
+            std::unique_ptr<typename ReturnPromiseType::Producer> completionProducer;
+            {
+                Locker lock { m_lock };
+                completionProducer = std::exchange(m_completionProducer, { });
+            }
```

Comment:
Suggestion, while you're touching this line: Instead of a default init + assignment, how about an immediately-invoked lambda? Something like `auto cP = [this]() { lock; return exchange(...); }();`